### ISSUE TITLE
Dockerize local environment - research

### ISSUE THREAD

**Issue by ktamaral** [pod4lib/aggregator#532]
As a developer, I would like to have a containerized local development environment for a quick and repeatable installation and setup process. Creating a dockerized local environment will take some upfront investment for research and implementation, but will reduce onboarding time for new developers and allow for the ability to make changes to the installation consistently across all environments.

Proposal:

- Research and evaluate the complexity of dockerizing the local environment
- The Harvard team has knowledge and experience with dockerizing RoR apps
  - Note the caveat that we aren’t yet sure of the complexity given no prior experience with POD
- The purpose of this proposal is to research the level of complexity to provide more information on whether to proceed with the work

Requirements:
- Dockerization of the application
- Dockerization of other components such as redis instance
- Scripts to populate test data in the local environment
- A docker compose file to orchestrate provisioning of the app and other components
- Instructions for setting up configuration and running the docker compose
- Docker must be optional, any changes made must not affect the manual setup process

**Comment by thatbudakguy** [comment]
I'd be curious to know what the benefits of running the application itself in a container are, given that the deployment environment is not containerized. In my experience, docker is usually used to achieve parity between environments, so having a dev setup that's different from staging/prod in this way doesn't seem optimal.

Re: running the dependent services in containers, I've been using this simple compose file:
```yml
services:
  db:
    image: postgres
    environment:
      POSTGRES_USER: pod4lib
      POSTGRES_PASSWORD: pod4lib
      POSTGRES_DB: aggregator
    ports:
      - "5432:5432"
    volumes:
      - db:/var/lib/postgresql/data

  cache:
    image: redis
    ports:
      - "6379:6379"

volumes:
  db:
```

following the instructions in the README to set environment variables like:
```sh
export REDIS_HOST=localhost
export REDIS_PORT=6379
export DATABASE_URL=postgresql://pod4lib:pod4lib@localhost:5432/aggregator
bundle exec rails server  # or other commands
```
has worked for me.

**Comment by thatbudakguy** [comment]
also in case it's helpful, here's [a `Dockerfile` that we used on a previous project](https://github.com/sul-dlss/dlme/blob/eaf8be093db5cbdc41ba833f16733c8417e52667/docker/Dockerfile) where we had a rails-based app with a worker/sidekiq instance, and wanted to build production and dev versions of both of those services. we used build targets to keep the dependencies separate for each service version.

**Comment by ktamaral** [comment]
Pull Request: https://github.com/pod4lib/aggregator/pull/550

**Comment by corylown** [comment]
Closed via #550
